File: Solutions/Medium/3315. Construct the Minimum Bitwise Array II/Solution.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        n: int = len(nums)
        result: List[int] = []
        for i in range(0, n):
            exp: int = 1
            res: int = -1
            while((exp & nums[i]) != 0):
                res = nums[i] - exp
                exp <<= 1
            result.append(res)
        return result

File: Solutions/Medium/3315. Construct the Minimum Bitwise Array II/Solution.cs
public class Solution {
    public int[] MinBitwiseArray(IList<int> nums) {
        int n = nums.Count;
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            int res = -1;
            int exp = 1;
            while ((nums[i] & exp) == 1) {
                res = nums[i] - exp;
                exp <<= 1;
            }
            result[i] = res;
        }
        return result;
    }
}